ovn-org/ovn-kubernetes
# Add support of HW backed devices in the full mode

**- What this PR does and why is it needed**
This series is a solution for #3159. It allows usage of HW backed devices, such as Virtual Functions (VFs) and Scalable Functions (SFs), and their representors as management ports of ovnkube to utilize HW offloading. Feature intended for usage only with mentioned devices, but not limited to them since ovnkube doesn't do any verification of the device type.
If netdevice and its representor names are specified in the user configuration and ovnkube is running in the full mode, then ovnkube will do the same configuration, as it is done now in the dpu-host and dpu modes: netdevice will be given IP address, routes configured, etc., and representor will be added to the OVS bridge, flows added to the datapath, etc.. Because two management port resides on the same node, netdevice management port renamed to `ovn-k8s-mp0`, its representor to `ovn-k8s-mp0_0`. If no device names were provided then ovnkube will use internal OVS port. Providing netdevice or representor name only is an error.

**- How to verify it**
- Unit-tests do check basic functionality;
- For manual testing: build regular k8s cluster with, for ex., Mellanox ConnectX-5 NIC in the worker node; create SRIOV VFs and enable switchdev mode; specify `OVNKUBE_NODE_MGMT_PORT_NETDEV` and `OVNKUBE_NODE_MGMT_PORT_REPRESENTOR` names in ovnkube-node worker yaml file (if workload pods with VFs are deployed - exclude specified VF id from sriov-network-device-plugin resource pool); deploy ovnkube-node on the worker node.

**- Description for the changelog**
Add support of HW backed devices in the full mode
